### DISCUSSION TITLE
error: TypeError: Self-contained binaries don't support module loading

### DISCUSSION THREAD

**Discussion by iwindx** [denoland/deno#11788] [Q&A]
I want to ask a question. I compiled the oak based web framework using compile API, and there was an error when I started it on liunx, and I didn't find a corresponding solution at all. I understand that compiling is packaged into binary, and all HTTP dependencies are added to the binary package. I'm a little confused about this. Hope to get a reply, thank you for taking the time to answer, thank you

**Comment by iwindx** [comment]
![image](https://user-images.githubusercontent.com/16376649/130203666-f7e67d34-fd64-4db2-a5f6-1b985ce112ee.png)
Attach a screenshot of the failure

**Comment by kitsonk** [comment]
Is some of the code trying to do a dynamic `import()` of modules? That is likely the cause.

**Reply by iwindx** [reply]
Thank you. The problem is solved
